Sources/PackageGraph/PackageModel+Extensions.swift

```diff
@@ -14,9 +14,9 @@ import SourceControl
 extension PackageDependencyDescription {
     /// Create the package reference object for the dependency.
     public func createPackageRef(config: SwiftPMConfig) -> PackageReference {
-        let effectiveURL = config.mirroredURL(forURL: self.url)
+        let effectiveURL = config.mirroredURL(forURL: url)
         return PackageReference(
-            identity: PackageReference.computeIdentity(packageURL: effectiveURL),
+            identity: explicitName?.lowercased() ?? PackageReference.computeIdentity(packageURL: effectiveURL),
```

Comment:
This seems like a change in semantics.  Are there any risks here?

Reply:
Good catch. If you remove the left-hand side of that `??` expression, tests will fail, with reports of mismatched dependency names. So although this looks like a semantic change, it's merely a downstream effect of consolidating `PackageGraphRoot.PackageDependency` into `PackageDependencyDescription`.

Later on, we will be changing the semantics so that we only use the package URL (normalized for scheme, case, etc.) for identity. Any explicit names would then become convenient labels rather than something load-bearing — more like a variable name within a block than a memory address.

Reply:
I see; thanks for the explanation.  If this doesn't have any actual effect, it would be great if there were a comment explaining what's going on.  If this will go away when the notion of package identity changes, then this seems OK for now.

Reply:
Agreed. Added with 0ccdeaf4.